Sources/MongoSwift/BSON/AnyBSONValue.swift

```diff
@@ -3,24 +3,46 @@ import Foundation
 /// A struct wrapping a `BSONValue` type that allows for encoding/
 /// decoding `BSONValue`s of unknown type.
 public struct AnyBSONValue: Codable, Equatable, Hashable {
-    // TODO: conform all `BSONValue` types to `Hashable` (SWIFT-320).
     public func hash(into hasher: inout Hasher) {
```

Comment:
the linter is complaining about "cyclomatic complexity" because there are so many branches in this function. There's no way to avoid the complexity when we have to switch like this, so you just need to add a swiftlint ignore for it. You can examples of that ignore elsewhere in the driver by grepping for "cyclomatic".